webapp/channels/src/components/post_view/reaction_list/reaction_list.tsx

```diff
@@ -78,15 +78,13 @@ export default class ReactionList extends React.PureComponent<Props, State> {
     }
 
     static getDerivedStateFromProps(props: Props, state: State): Partial<State> | null {
-        let emojiNames = state.emojiNames;
+        const emojiNames = Object.values(props.reactions ?? {}).map((reaction) => reaction.emoji_name);
```

Comment:
What are the changes to this method for? As far as I can tell, it looks like it should behave similarly to before, except we're now checking if the value of `emojiNames` is shallowly equal to the previous version instead of assuming that the array is immutable and just checking if the reference is the same as before or not

Reply:
The changes ensures that `emojinames` is always up to date with the reactions and matches it. I noticed that it was not updating when the max emoji number was reached except I refresh the page